### PR TITLE
feat(insights): add hits and attributes to InsightsEvent

### PR DESCRIPTION
**Pull request by eunjae-lee**
**Summary**

This PR adds `hits` and `attributes` to InsightsEvent.

## Detail

Previously we had this:
```
export type InsightsEvent = {
  insightsMethod?: InsightsClientMethod;
  payload: any;
  widgetType: string;
  eventType: string;
};
```

For example,

- insightsMethod: "viewedObjectIDs"
- widgetType: "ais.hits"
- eventType: "view"

`window.aa(insightsMethod, payload);` just worked, which means `payload` is a tailored object for Algolia Insights.

When users want to send events to third party trackers, they might need more information other than what is in the `payload`.

In case of `Hits Viewed`, `payload` contains `eventName`, `index` and `objectIDs`. When they need to send events to, for example, Segment, they need other data like position, product name, etc.

So we're passing the whole `hits` object inside the `InsightsEvent`.

In case of facet widgets, we're adding `attribute` to `InsightsEvent`. It's to easily filter out events like:

```
onEvent: (event, aa) {
  if (event.attribute === 'price') {
    // do not track events for `price` attribute
  }
  aa(event.insightsMethod, event.payload);
}
```

### CODE REVIEW

File: src/lib/utils/createSendEventForHits.ts [hunk 1]
```diff
@@ -60,7 +60,8 @@ const buildPayload: BuildPayload = ({
       return null;
     }
   }
-  const hitsArray = Array.isArray(hits) ? hits : [hits];
+  // `Array.from(hits)` is to remove `hits.__escaped`
+  const hitsArray = Array.isArray(hits) ? Array.from(hits) : [hits];
```
Comment (Haroenv) [inline src/lib/utils/createSendEventForHits.ts:64]: Array.from isn't compatible with IE

Comment (eunjae-lee) [reply src/lib/utils/createSendEventForHits.ts]: ah right. https://github.com/algolia/instantsearch.js/pull/4667/commits/d3868b519395a21ca95d9b239206d8265ba0669e

Comment (Haroenv) [reply src/lib/utils/createSendEventForHits.ts]: you should instead remove the property if you don't want it to show up, or map over it

Comment (eunjae-lee) [reply src/lib/utils/createSendEventForHits.ts]: https://github.com/algolia/instantsearch.js/pull/4667/commits/3b541f54c5ea3d8a41f6f5e9b3fac18a6a7f8530 WDTY?

**Review by Haroenv — APPROVED** [review]
I see why it's not in payload, since that would cause the default implementation to need to strip them, but I kinda feel like it fits better under payload or metadata than as top-level key

**Comment by Haroenv** [comment]
(looks like some tests have failed due to the delete instead of spreading)

**Comment by eunjae-lee** [comment]
> I see why it's not in payload, since that would cause the default implementation to need to strip them, but I kinda feel like it fits better under payload or metadata than as top-level key

you're right. `attribute` could probably be there as a top-level key, but `hits` doesn't really fit much. Do you think it makes sense to have a key like `metadata` and have `attribute` or `hits` under it?

**Pull request merged**